src/Umbraco.Core/Logging/DisposableTimer.cs

```diff
@@ -53,14 +53,21 @@ public class DisposableTimer : DisposableObjectSlim
                     case LogLevel.Debug:
                         if (startMessageArgs == null)
                         {
-                            logger.LogDebug("{StartMessage} [Timing {TimingId}]", startMessage, _timingId);
+                            if (logger.IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+                            {
+                                logger.LogDebug("{StartMessage} [Timing {TimingId}]", startMessage, _timingId);
+                            }
                         }
                         else
                         {
-                            var args = new object[startMessageArgs.Length + 1];
-                            startMessageArgs.CopyTo(args, 0);
-                            args[startMessageArgs.Length] = _timingId;
-                            logger.LogDebug(startMessage + " [Timing {TimingId}]", args);
+                            if (logger.IsEnabled(Microsoft.Extensions.Logging.LogLevel.Debug))
+                            {
+                                var args = new object[startMessageArgs.Length + 2];
+                                startMessageArgs.CopyTo(args, 0);
+                                args[startMessageArgs.Length - 1] = startMessage;
+                                args[startMessageArgs.Length] = _timingId;
+                                logger.LogDebug("{StartMessage} [Timing {TimingId}]", args);
```

Comment:
Thanks @nzdev for this PR.

I have been checking and testing it, and unfortunately I think it is not quite working as expected. 

There seems to be a global issue with the indexes at the different updated parts.
For the specific lines here for example:

- either we go with startMessageArgs, but then I think we have to use 
```
    args[startMessageArgs.Length] = startMessage;
    args[startMessageArgs.Length + 1] = _timingId;
```

- either we go for args itself and then we have to use

```
    args[args.Length - 2] = startMessage;
    args[args.Length -1] = _timingId;
```

With the current situation for example, we actually overwrite the last arg copied from startMessageArgs by the message itself : args[startMessageArgs.Length - 1] = startMessage

Also, I'm not sure this approach is working correctly, as the logging message format is not output as expected. In your case "{StartMessage}" is seen as the first argument to use from args (and is not replaced by the full original message), while in the original code we append the message as is, with its own arguments formatted.

I tried a basic test with the following:
- message => "Determining hash of code files on disk {DiskName}"
- args => "coucou"

The result with original code is as expected:
    Determining hash of code files on disk "coucou" [Timing "dc9a8f9"]

The result with the updated code is not as expected:
    "coucou" [Timing "Determining hash of code files on disk {DiskName}"]
    

I hope my explanations make sense, let me know if they don't 😉

Reply:
Hello @nzdev , no hurry but I just wanted to make sure you had seen the remark above about the indexes, since you reacted to the other one already 😁

Reply:
Thanks @mikecp . Haven't had a chance yet to wrap my head around this one. Will likely be late this week before I get a chance.